src/AutoRest.CSharp/Common/Input/InputTypes/InputParameter.cs

```diff
@@ -45,7 +45,75 @@ internal record InputParameter(
         SkipUrlEncoding: false,
         Explode: false,
         ArraySerializationDelimiter: null,
-        HeaderCollectionPrefix: null,
-        SerializationFormat: SerializationFormat.Default)
+        HeaderCollectionPrefix: null)
     { }
+
+    internal InputParameter(
+        string name,
+        string nameInRequest,
+        string? description,
+        InputType type,
+        RequestLocation location,
+        InputConstant? defaultValue,
+        VirtualParameter? virtualParameter,
+        InputParameter? groupedBy,
+        InputOperationParameterKind kind,
+        bool isRequired,
+        bool isApiVersion,
+        bool isResourceParameter,
+        bool isContentType,
+        bool isEndpoint,
+        bool skipUrlEncoding,
+        bool explode,
+        string? arraySerializationDelimiter,
+        string? headerCollectionPrefix,
+        SerializationFormat serializationFormat) : this(
+            name,
+            nameInRequest,
+            description,
+            type,
+            location,
+            defaultValue,
+            virtualParameter,
+            groupedBy,
+            kind,
+            isRequired,
+            isApiVersion,
+            isResourceParameter,
+            isContentType,
+            isEndpoint,
+            skipUrlEncoding,
+            explode,
+            arraySerializationDelimiter,
+            headerCollectionPrefix)
+    {
+        _serializationFormat = serializationFormat;
+    }
+
+    private SerializationFormat? _serializationFormat;
+    public SerializationFormat SerializationFormat => _serializationFormat ??= GetSerializationFormat(Type, Location);
+
+    private static SerializationFormat GetSerializationFormat(InputType parameterType, RequestLocation requestLocation)
```

Comment:
When a InputParameter is converted from codeModel(modelerFour), the _serializationFormat is not null, this function will not be called. We may need the strategy for parameter from swagger, also.

Reply:
This is not the purpose. See the comment below